linode/manager
# poc: Update from React `17` to `18`

## Description 📝

Putting this POC up to investigate what remains from updating from React 17 to 18

> [!note]
> We can choose to proceed with this PR, but I mostly intend for this to just be a quick POC so we know what is required to perform the update

## Changes  🔄
- Had to re-generate the `yarn.lock`
- Luxon types changed for some reason
- Had to update `react-router-dom` to fix types
- Had to update some Launch Darkly imports
- For some reason, `placeholder` does not exist on a native `<select>`? I'm confused about this one
- Performed many various fixes and adjustments to account for new types
- React Testing Library needs to be updated to >= 13 to use React 18 (if we don't update it, our code will run as if it was React 17 I believe)

## How to test 🧪

### Prerequisites
- Check out this PR
- Run `yarn` in the repo's root for good measure to make sure your dependencies are installed

### Verification steps 
- Verify all CI and Cypress tests pass
- Verify a production build of Cloud Manager works (you can check with the preview link)
- Verify that the local dev server works like normal (`yarn dev`)

## As an Author I have considered 🤔
